spyder/widgets/sourcecode/codeeditor.py

```diff
@@ -2698,10 +2698,9 @@ def keyPressEvent(self, event):
                    and self.codecompletion_enter:
                     self.select_completion_list()
                 else:
-                    cmt_or_str = self.in_comment_or_string()
                     self.textCursor().beginEditBlock()
                     TextEditBaseWidget.keyPressEvent(self, event)
-                    self.fix_indent(comment_or_string=cmt_or_str)

```

Comment:
The problem with deactivating this is that indentation would become active for when you're working inside block comments. That's why I added this.

Instead we should be smarter and check if the line *doesn't* start with a comment and *ends* with it one to apply indentation in that case.